actionpack/lib/action_dispatch/http/parameters.rb

```diff
@@ -81,6 +81,17 @@ def path_parameters
         get_header(PARAMETERS_KEY) || set_header(PARAMETERS_KEY, {})
       end
 
+      # Returns true if parameters are parse-able. Otherwise returns false.
+      #
+      #   GET /posts/5?x=1&y=2            | request.params_readable? => true
+      #   GET /posts/5?x[y]=1&x[y][][w]=2 | request.params_readable? => false
+      #
+      def params_readable?
+        !!parameters
```

Comment:
Oh right, I recall the really long discussion about the double `!` operator... We will change that to just `parameters`.

(Although I personally don't think it's a good thing to keep doing)

c/ @xtina-starr